src/main/java/ch/njol/skript/effects/EffLeash.java

```diff
@@ -0,0 +1,80 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Leash entities")
+@Description("Leash living entities to other entities")
+@Examples("leash the player to the target entity")
+@Since("INSERT VERSION")
+public class EffLeash extends Effect {
+
+	static {
+		Skript.registerEffect(EffLeash.class,
+				"(leash|lead) %livingentities% to %entity%",
+				"make %entity% (leash|lead) %livingentities%",
+				"un(leash|lead) [holder of] %livingentities%");
+	}
+	
+	@SuppressWarnings("null")
+	private Expression<Entity> holder;
+	@SuppressWarnings("null")
+	private Expression<LivingEntity> targets;
```

Comment:
Since you added unleash this should be @Nullable

Reply:
a boolean can't be null, it's default value is always false.

Reply:
this comment is about the line above